arrow/src/ipc/writer.rs

```
@@ -175,13 +186,37 @@ impl IpcDataGenerator {
                     )?;
                 }
             }
+            _ => (),
```

maybe this should return an error for unsupported nested types

---

Hmm, I guess that it doesn't just returning an error, is because if no dictionaries exist inside the fields of these nested types, then the ipc writer still can work. To return an error for dictionaries existing inside unsupported nested types, we need to look into them as supported types, so it's basically almost near to implement them.

I may take some time continuing the work to add more supports (map, etc) here.